src/lib/api/patrons/patron.js

```diff
@@ -14,17 +16,63 @@ const get = async patronPid => {
   return response;
 };
 
-const list = async queryText => {
-  const response = await http.get(`${listUrl}?q=${queryText}*`);
+const list = async (queryText, wildcard = true) => {
+  const response = await http.get(
+    `${listUrl}?q=${queryText}${wildcard ? '*' : ''}`
+  );
   response.data.total = response.data.hits.total;
   response.data.hits = response.data.hits.hits.map(hit =>
     serializer.fromJSON(hit)
   );
   return response;
 };
 
+class QueryBuilder {
+  constructor() {
+    this.patronQuery = [];
+    this.emailQuery = [];
+    this.nameQuery = [];
+    this.patronUniqueIDQuery = [];
+  }
+
+  withEmail(email, wildcard = false) {
+    if (!email) {
+      throw TypeError('Email argument missing');
+    }
+    this.patronQuery.push(`email:${email}${wildcard ? '*' : ''}`);
+    return this;
+  }
+
+  withName(name, wildcard = false) {
+    if (!name) {
+      throw TypeError('Name argument missing');
+    }
+    this.patronQuery.push(`name:${name}${wildcard ? '*' : ''}`);
+    return this;
+  }
+
+  withPatronUniqueID(patronID) {
```

Comment:
In my opinion, we cannot put this here... because when I see it I think than that the PID or ID are not unique...